### PR TITLE
Feature/elyard/controller implementation

### PR DESCRIPTION
**Pull request by danielmohansahu**

### CODE REVIEW

File: app/Limits.cpp [hunk 1]
```diff
@@ -17,12 +17,99 @@ Limits::Limits(const std::shared_ptr<const Params>& params)
   : params_(params) {
 }
 
+double Limits::throttleToSpeed(const double throttle) const {
+  double speed_calc;
+  if (throttle < 0) {
+    speed_calc = throttle * params_->velocity_min;
```
Comment (danielmohansahu) [inline app/Limits.cpp:23]:
```suggestion
    speed_calc = std::abs(throttle) * params_->velocity_min;
```

File: app/Limits.cpp [hunk 2]
```diff
@@ -17,12 +17,99 @@ Limits::Limits(const std::shared_ptr<const Params>& params)
...
+  }  else {
+    speed_calc = throttle * params_->velocity_max;
+  }
+  return speed_calc;
+}
+
+double Limits::speedToThrottle(const double speed) const {
+  double throttle_calc;
+  if (speed < 0) {
+    throttle_calc = - (speed / params_->velocity_min);
```
Comment (danielmohansahu) [inline app/Limits.cpp:33]: I think this should be an error case; `speed` should never be negative. Should this function be called `velocityToThrottle`?

Comment (selyard) [reply app/Limits.cpp:33]: Will change to a floor at 0 so worst case negative values cause rover to stop.

File: app/Limits.cpp (hunk 1 above)
Comment (selyard) [reply app/Limits.cpp:23]: Will change to a floor at 0 so worst case negative values cause rover to stop.

**Pull request closed without merging**